pkg/authorization/registry/rolebinding/policybased/virtual_storage.go

```diff
@@ -272,7 +272,7 @@ func (m *VirtualStorage) confirmNoEscalation(ctx kapi.Context, roleBinding *auth
 	ownerRightsCover, missingRights := rulevalidation.Covers(ownerRules, modifyingRole.Rules)
 	if !ownerRightsCover {
 		user, _ := kapi.UserFrom(ctx)
-		return fmt.Errorf("attempt to grant extra privileges: %v\nuser=%v\nownerrules%v\n", missingRights, user, ownerRules)
+		return kapierrors.NewUnauthorized(fmt.Sprintf("attempt to grant extra privileges: %v\nuser=%v\nownerrules%v\n", missingRights, user, ownerRules))
```

Comment:
this is probably fine for now... would be nice to have the gorpy bits in Details fields and a nice summary in the message, but we don't have anything that can reduce rules to a pretty string